sys-apps/kexec-tools/files/kexec-auto-load

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bash
+
+# Defaults
+BOOTPART=/boot
+KNAME=kernel
+INITRD=
+
+if [[ -f /etc/kernel/cmdline ]]; then
+	KPARAM="$(tr -s "${IFS}" ' ' </etc/kernel/cmdline)"
+elif [[ -f /usr/lib/kernel/cmdline ]]; then
+	KPARAM="$(tr -s "${IFS}" ' ' </usr/lib/kernel/cmdline)"
+else
+	KPARAM=
+fi
+
+instkern_log=/var/log/installkernel.log
+if [[ -f ${instkern_log} ]]; then
```

Comment:
If you use `-s` here instead of `-f`, then empty logfiles will be ignored as well, which might come from logrotate having rotated the log prior to the next installkernel run

Reply:
Having said that. What would be do in general, if logrotate would rotate that file?

Now I am thinking that maybe some sort of 'state file' would be better that is totally under installkernel control, so for example just move it to something like `/var/lib/installkernel/`

Some people will have something like "*.log" in their logrotate.conf, which would pretty much break this apporach or am I missing something here?

Reply:
To be honest, I have no clue what logrotate will do here since I don't use it myself.

I assumed that logrotate would never leave a completely empty log file behind, i.e. that when rotating it would leave the last x lines behind or something like that. Is that not how it works? Can we somehow configure it to not leave the log completely empty?

Reply:
After having looked into it, I think this logrotate config snippet should resolve the problem, it ensures that the header as well as the last installed kernel is always kept:

```
cat /etc/logrotate.d/installkernel
/var/log/installkernel.log {
     delaycompress
     create
     missingok
     notifempty
     postrotate
         head -n1 $2 >> $1
         tail -n1 $2 >> $1
     endscript
}
```

What do you think? Do we need to add something else to this rule?

Reply:
In the end its up to the user how to configure log rotation. And I think ist not unusual to have additional rules that just rotate everything in matching `/var/log/*.log` and then touch an empty file so that the correct perms and owner can be set for daemons that ar not running as root.

Files in /var/log are assumes to be logfiles, which means they normally only grow and that they contain only logs.

Having a file here which must noch be empty and must not be rotated by other means that with a specific logrotate-config seems at least a bit odd to me.

TBH I would choose another location for such kind of state information as this is not really a log as we normally find here.

Reply:
I'm not sure I follow. Many other packages install config snippets into `/etc/logrotate.d` and the comment in `/etc/logrotate.conf` also says that this is the specific purpose of that directory. 

Any user added rules should be above the `include /etc/logrotate.d` line in `/etc/logrotate.conf` so blanket rules don't override the snippets installed by the packages. You could instead put the the blanket rule after this line, and then it will override the package installed config snippets, but doing so is a terrible idea regardless of installkernel.log. You'd end up overriding, for example, also the config snippets for nginx and mysql, both of which contain `postrotate` scripts that are important for the proper functioning of nginx and mysql.

I'm open to installing this log somewhere else, but I'd like to avoid having two files doing basically the same thing (one log and one state file). And I honestly don't really see the point since blindly overriding all config snippets installed by packages is a terrible idea regardless.

Reply:
Yes, but I never saw anywhere, that it was required for a log file to contain any information so a part of the system would work. I know that there are indeed peopple out there that just create new rules (be it clever or not) that apply another (maybe shorter) retention time for all log files by using a pattern like `*.log`

I would say you cannot rely on a log file under `/var/log` to contain any lines or not to be rotated by other means than your specific logrotate-config snippet.

It may feel a little too complicated to additioanally create a new state file, but for a clean design and - most important - robustness, I would favor of doing it.

Otherwise I would expect users to complain that installkernel/kexec update would silently fail at some point because the installkernel.log is empty.

Reply:
Alright, your argument makes sense. Added a state file in installkernel v46.